### ISSUE TITLE
add methods for arithmetic operators?

### ISSUE THREAD

**Issue by Crghilardi** [evetion/GeoArrays.jl#13]
I've been working on a project where I calculate two `GeoArray` layers using `PointCloudRasterizers` and then calculate a difference. 
I can use `layer1 - layer2` syntax and it calculates successfully but converts to a standard `Array` type and casting it as a `GeoArray` afterwards resets the coordinate information. 
It would be nice to be able to do band math operations and have the coordinate info and everything get passed through automatically. 
My example is simple and I know all layers are the same info, so I have not thought through how difficult it becomes if there are differences.If someone can point me to an example or give me some guidance on what this might look like I can try and put it together for a PR.

```julia
#example from PointCloudRasterizers README,  file from GeoArrays readme doesn't exist?

lazfn = joinpath(dirname(pathof(LazIO)), "..", "test/libLAS_1.2.laz")
pointcloud = LazIO.open(lazfn)
cellsizes = (1.,1.)
raster_index = index(pointcloud, cellsizes)

raster_maximum = reduce(raster_index, field=:Z, reducer=maximum)
raster_minimum = reduce(raster_index, field=:Z, reducer=minimum)

#original affine info from raster_minimum.f
AffineMap([1.0 0.0; 0.0 -1.0], [375000.03, 380000.03])

#calculate difference layer
raster_range = raster_maximum - raster_minimum #5000×5000×1 Array{Union{Missing, Float64},3}
#just casting it to GeoArray resets coord info
GeoArray(raster_range).f #AffineMap([1.0 0.0; 0.0 1.0], [0.0, 0.0])

#I can manually construct it afterwards, but that seems very manual
GeoArray(raster_range,raster_minimum.f,"")

```

**Comment by evetion** [comment]
This would be a nice feature. Short version:

**Now**
```julia
GeoArray(rand(5, 5)) - GeoArray(rand(5, 5))  # 5×5×1 Array{Float64,3}
```

**Future**
```julia
GeoArray(rand(5, 5)) - GeoArray(rand(5, 5))  # 5×5×1 GeoArray{Float64,3}
```

However, this can only be allowed to work on `GeoArray`s that have the same `AffineMap`and `crs`, otherwise it should `error` out for now. In the future we can support operations while translating based on different `AffineMap`s or even warping in the case of different `crs`s.

Feel free to make a PR, should be easy to override `Base` methods. I would add them in a new `operations.jl` file. Which operators do you want to support?

Small example:
```julia
function Base.:-(a::GeoArray, b::GeoArray)
    a.A - b.A  # and some more stuff
end
```

**Comment by evetion** [comment]
> However, this can only be allowed to work on GeoArrays that have the same AffineMapand crs, otherwise it should error out for now.

They also need to be the same size.

**Comment by evetion** [comment]
Fixed by #19. Thanks for using and extending the package! 🥇
